spec/lib/zendesk_apps_tools/api_connection_spec.rb

```diff
@@ -0,0 +1,116 @@
+require 'spec_helper'
+require 'api_connection'
+
+describe ZendeskAppsTools::APIConnection do
+  let(:subdomain_validation_pattern) { ZendeskAppsTools::APIConnection::SUBDOMAIN_VALIDATION_PATTERN }
+  let(:url_validation_pattern)       { ZendeskAppsTools::APIConnection::ZENDESK_URL_VALIDATION_PATTERN }
+  let(:default_url_template)         { ZendeskAppsTools::APIConnection::DEFAULT_URL_TEMPLATE }
+
+  describe 'CONSTANTS' do
+    describe 'DEFAULT_URL_TEMPLATE' do
+      context '% subdomain (used in private method full_url)' do
+        it 'replaces %s with subdomain in template' do
+          user_input_subdomain = 'my-subdomain'
+          expect(default_url_template % user_input_subdomain).to eq("https://my-subdomain.zendesk.com/")
+        end
+      end
+    end
+
+    describe 'SUBDOMAIN_VALIDATION_PATTERN' do
+      context 'valid_subdomain? (a private method)' do
+        define_method(:valid_subdomain?) { |subdomain| !!subdomain_validation_pattern.match(subdomain) }
+
+        it 'returns false if subdomain is NOT in valid format' do
+          expect(valid_subdomain?('sub.domain')).to eq(false)
+          expect(valid_subdomain?('sub!domain')).to eq(false)
+          expect(valid_subdomain?('sub~domain')).to eq(false)
+          expect(valid_subdomain?('sub_domain')).to eq(false)
+        end
+
+        it 'returns true if subdomain is in valid format' do
+          expect(valid_subdomain?('subDomain')).to eq(true)
+          expect(valid_subdomain?('SUBDOMAIN')).to eq(true)
+          expect(valid_subdomain?('subdomain')).to eq(true)
+          expect(valid_subdomain?('sub-domain')).to eq(true)
+        end
+      end
+    end
+
+    describe 'ZENDESK_URL_VALIDATION_PATTERN' do
+      context 'valid_full_url? (a private method)' do
+        define_method(:valid_full_url?) { |subdomain|  !!url_validation_pattern.match(subdomain) }
+
+        context 'with regular zendesk urls' do
+          it 'returns false when subdomain does not match full url pattern' do
```

Comment:
I'd ideally like to see these register as valid as well

Reply:
👍